api/envoy/extensions/filters/http/admission_control/v3alpha/admission_control.proto

```diff
@@ -91,4 +91,13 @@ message AdmissionControl {
   // below this threshold, rejection probability will increase. Any success rate above the threshold
   // results in a rejection probability of 0. Defaults to 95%.
   config.core.v3.RuntimePercent sr_threshold = 5;
+
+  // If the number of requests in the last second is less than this threshold, the request
+  // will not be rejected, even if the success rate is lower than sr_threshold.
+  // Defaults to 0.
+  config.core.v3.RuntimeUInt32 rps_threshold = 6;
```

Comment:
I don't think this should only consider the last 1 second. It would make more sense to look at something like the minimum number of samples in the sampling window. It would still accomplish the same thing, but give better behavior for bursty request patterns.

Would that work for your particular use-case?

Reply:
To be more clear, you can still represent what I'm suggesting as an RPS threshold. It would just be an "average RPS" threshold across the entirety of the sampling window.

Reply:
Average RPS is indeed more reasonable. How about calculating it over a fixed period of time but not the whole sampling window? Since the sampling window is configurable and it might be a long time window.